src/gui/properties/trackerlist.cpp

```diff
@@ -529,13 +543,61 @@ void TrackerList::showTrackerListMenu(QPoint) {
 
 void TrackerList::loadSettings()
 {
-    if (!header()->restoreState(Preferences::instance()->getPropTrackerListState())) {
-        setColumnWidth(0, 30);
-        setColumnWidth(1, 300);
-    }
+    header()->restoreState(Preferences::instance()->getPropTrackerListState());
 }
 
 void TrackerList::saveSettings() const
 {
     Preferences::instance()->setPropTrackerListState(header()->saveState());
 }
+
+QStringList TrackerList::headerLabels()
+{
+    QStringList header;
+    header << "#";
+    header << tr("URL");
+    header << tr("Status");
+    header << tr("Received");
+    header << tr("Seeds");
+    header << tr("Peers");
+    header << tr("Downloaded");
+    header << tr("Message");
+
+    return header;
+}
+
+void TrackerList::displayToggleColumnsMenu(const QPoint&)
+{
+    QMenu hideshowColumn(this);
+    hideshowColumn.setTitle(tr("Column visibility"));
+    QList<QAction*> actions;
```

Comment:
Instead of using the additional list with actions, you can put the index inside of the action object:
```C++
QAction *myAct = hideshowColumn.addAction(headerLabels().at(i));
myAct->setCheckable(true);
myAct->setChecked(!isColumnHidden(i));
myAct->setData(i);
```
and later instead of 
```C++
int col = actions.indexOf(act);
```
you would do:
```c++
int col = act->data().toInt();
```